File: Leetcode/58-length-of-last-word/Solution.cc
class Solution {
public:
    int lengthOfLastWord(string s) {
        if(s.size() == 0)       return 0;
        int i = s.size() - 1;
        // skip the spaces
        while(i >= 0 && s[i] == ' ')    i--;
        int end = i;
        while(i >= 0 && s[i] != ' '){
            i--;
        }
        return end - i;
    }
};

File: Leetcode/58-length-of-last-word/Solution.java
public class Solution {
    public int lengthOfLastWord(String s) {
        if(s.length() == 0)       
            return 0;
        int i = s.length() - 1;
        // skip the trailing spaces
        while(i >= 0 && s.charAt(i) == ' ')    i--;

        int end = i;
        while(i >= 0 && s.charAt(i) != ' '){
            i--;
        }
        return end - i;
    }
}